lib/HLSL/HLOperationLower.cpp

```diff
@@ -5670,7 +5670,20 @@ Value *TranslateAllocateRayQuery(CallInst *CI, IntrinsicOp IOP,
                                  HLObjectOperationLowerHelper *pObjHelper,
                                  bool &Translated) {
   hlsl::OP *hlslOP = &helper.hlslOP;
-  Value *refArgs[] = {nullptr, CI->getOperand(1)};
+  // upgrade to allocateRayQuery2 if there is a non-zero 2nd template arg
```

Comment:
In this location, you might say if RayQueryFlags are non-zero, then add indices to `HLOperations.h` and use them, like:
```cpp
// AllocateRayQuery
const unsigned kAllocateRayQueryRayFlagsIdx = 1;
const unsigned kAllocateRayQueryRayQueryFlagsIdx = 2;
```